test/integration/ddl_sharding_key_test.lua

```diff
@@ -0,0 +1,568 @@
+
+local t = require('luatest')
+local crud = require('crud')
+
+local helpers = require('test.helper')
+
+local ok = pcall(require, 'ddl')
+if not ok then
+    t.skip('Lua module ddl is required to run test')
+end
+
+local pgroup = helpers.pgroup.new('ddl_sharding_key', {
+    engine = {'memtx', 'vinyl'},
+})
+
+pgroup:set_before_all(function(g)
+    g.cluster = helpers.Cluster:new({
+        datadir = fio.tempdir(),
+        server_command = helpers.entrypoint('srv_ddl'),
+        use_vshard = true,
+        replicasets = helpers.get_test_replicasets(),
+        env = {
+            ['ENGINE'] = g.params.engine,
+        },
+    })
+    g.cluster:start()
+    local result, err = g.cluster.main_server.net_box:eval([[
+        local ddl = require('ddl')
+
+        local ok, err = ddl.get_schema()
+        return ok, err
+    ]])
+    t.assert_equals(type(result), 'table')
+    t.assert_equals(err, nil)
+end)
+
+pgroup:set_after_all(function(g) helpers.stop_cluster(g.cluster) end)
+
+pgroup:set_before_each(function(g)
+    helpers.truncate_space_on_cluster(g.cluster, 'customers_name_key')
+    helpers.truncate_space_on_cluster(g.cluster, 'customers_name_key_uniq_index')
+    helpers.truncate_space_on_cluster(g.cluster, 'customers_name_key_non_uniq_index')
+    helpers.truncate_space_on_cluster(g.cluster, 'customers_age_key')
+end)
+
+local function check_get(g, space_name, id, name)
+    local result, err = g.cluster.main_server.net_box:call('crud.get', {
+        space_name, {id, name}
+    })
+
+    t.assert_equals(err, nil)
+    t.assert_not_equals(result, nil)
+end
+
+pgroup:add('test_insert_object_get', function(g)
+    -- insert_object
+    local result, err = g.cluster.main_server.net_box:call(
+        'crud.insert_object', {'customers_name_key', {id = 1, name = 'Fedor', age = 59}})
+
+    t.assert_equals(err, nil)
+    t.assert_equals(result.metadata, {
+        {is_nullable = false, name = 'id', type = 'unsigned'},
+        {is_nullable = false, name = 'bucket_id', type = 'unsigned'},
+        {is_nullable = false, name = 'name', type = 'string'},
+        {is_nullable = false, name = 'age', type = 'number'},
+    })
+    local objects = crud.unflatten_rows(result.rows, result.metadata)
+    t.assert_equals(objects, {{id = 1, bucket_id = 86, name = 'Fedor', age = 59}})
+
+    -- get
+    local result, err = g.cluster.main_server.net_box:call('crud.get', {'customers_name_key', {1, 'Fedor'}})
+
+    t.assert_equals(err, nil)
+    t.assert(result ~= nil)
+    local objects = crud.unflatten_rows(result.rows, result.metadata)
+    t.assert_equals(objects, {{age = 59, bucket_id = 86, id = 1, name = "Fedor"}})
+
+    -- insert_object again
+    local obj, err = g.cluster.main_server.net_box:call(
+        'crud.insert_object', {'customers_name_key', {id = 1, name = 'Alexander', age = 37}})
+
+    t.assert_equals(err, nil)
+    t.assert_equals(obj.rows, {{1, 1690, "Alexander", 37}})
+end)
+
+pgroup:add('test_insert_get', function(g)
```

Comment:
Could we add a case where primary index is a json-path index?

Reply:
Why do you think it is worth to do it?

Reply:
submitted a separate issue about it -
https://github.com/tarantool/crud/issues/219